src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7942,7 +8100,54 @@ static int M(ReadOnlySpan<char> chars)
         }
 
         [Fact]
-        public void PatternMatchReadOnlySpanCharOnConstantStringCSharp_8()
+        public void PatternOrSwitchReadOnlySpanChar_MissingLength()
+        {
+            var sourceA =
+@"namespace System
+{
+    public ref struct ReadOnlySpan<T>
+    {
+        public ReadOnlySpan(T[] array) { }
+    }
+    public static class MemoryExtensions
+    {
+        public static ReadOnlySpan<char> AsSpan(string s) => default;
+        public static bool SequenceEqual<T>(ReadOnlySpan<T> a, ReadOnlySpan<T> b) => false;
+    }
+}";
+            var comp = CreateCompilation(sourceA);
+            var refA = comp.EmitToImageReference();
+
+            var sourceB =
+@"using System;
+class Program
+{
+    static void Main()
+    {
+        var s = new ReadOnlySpan<char>(new char[0]);
+        _ = s is ""str"";
```

Comment:
I'm fine with this scenario not working, but curious if there was a specific reason we can't compile this when 'Length' is missing? Seems like we only actually need to call MemoryExtensions.SequenceEqual?